pkg/apis/agones/v1/gameserver_test.go

```diff
@@ -319,8 +326,15 @@ func TestGameServerApplyDefaults(t *testing.T) {
 		},
 	}
 
+	runtime.FeatureTestMutex.Lock()
```

Comment:
This locks it from changing, but the test doesn't set it to an explicit value. Should the test

1. Lock the mutex
2. Capture the current value
3. Set a specific value for the test
4. Defer resetting the current value and unlocking the mutex

?

Reply:
The value for the Flags gets reset and set the appropriate value for the test here:
https://github.com/googleforgames/agones/pull/1324/files/641016f2ef479806519854856adb95150ee04007#diff-71e2ed803fe1fca8402cc9afb9117956R335

`ParseFeatures` will reset the entire feature flag set on each pass through, so on any test that requires an explicit Feature Flag setup, then `ParseFeatures` should be all that's required.

I'm also okay with the flags remaining post test - if a test fails because of a different Feature Flag than expected - then it should be explicitly setting that value, and seems like a valid test failure. Do you feel like it should return to a default state at the end of a test?

Does that cover the above?

Reply:
Yes, thanks for clarifying.